### ISSUE TITLE
Document error(transparent) attribute

### ISSUE THREAD

**Issue by dtolnay** [dtolnay/thiserror#55]
Added in https://github.com/dtolnay/thiserror/releases/tag/1.0.7. I am looking to build just a little more experience with this attribute in real code before settling on how best to approach it in docs.

**Comment by sunjay** [comment]
I was using `#[error("{0}")]` everywhere and I just replaced that with `#[error(transparent)]`. What do you think of allowing `#[error(...)]` on the struct itself as a sort of "default" format for any enum variant that doesn't specify the attribute?

For example, if you had the following code:

```rust
#[derive(Debug, Error)]
enum RenderError {
    #[error(transparent)]
    CanvasError(#[from] CanvasError),
    #[error(transparent)]
    DrawError(#[from] DrawError),
    #[error(transparent)]
    ScaleError(#[from] ScaleError),
}
```

You could turn it into:

```rust
#[derive(Debug, Error)]
#[error(transparent)]
enum TopError {
    CanvasError(#[from] CanvasError),
    DrawError(#[from] DrawError),
    ScaleError(#[from] ScaleError),
}
```

By treating the top `#[error(...)]` attribute as the default, I could then override one of the variants with something custom if I wanted to. By making this an additional behaviour of the attribute, rather than the default behaviour, this should be a backwards compatible change.

**Comment by dtolnay** [comment]
It works that way already I believe.

**Comment by sunjay** [comment]
You're right. Thanks! This crate has saved me a ton of boilerplate :smile:

**Comment by dtolnay** [comment]
Fixed in 1cce98b133e296631a04617ed7f043ac7cfdeea8.
